test/test_conversion_driver_esm1p5.py

```diff
@@ -106,33 +114,48 @@ def _mock_process_with_exception(error_message):
     yield _mock_process_with_exception
 
 
-
 @pytest.mark.parametrize(
-    "input_list", [[], ["fake_file"], ["fake_file_1", "fake_file_2", "fake_file_3"]]
+    "input_list", [[], ["fake_file"], [
+        "fake_file_1", "fake_file_2", "fake_file_3"]]
 )
-def test_convert_fields_file_list(mock_process, input_list):
-    esm1p5_convert.convert_fields_file_list(input_list, "fake_nc_write_dir")
+def test_convert_fields_file_list_success(mock_process, input_list):
+    input_list_paths = [Path(p) for p in input_list]
+    succeeded, _ = esm1p5_convert.convert_fields_file_list(
+        input_list_paths, "fake_nc_write_dir")
 
     assert mock_process.call_count == len(input_list)
 
+    successful_input_paths  = [successful_path_pair[0] for 
+                                successful_path_pair in succeeded]
+    
+    assert input_list_paths == successful_input_paths
+
 
-def test_convert_fields_file_list_excepted_error(mock_process_with_exception):
+def test_convert_fields_file_list_fail_excepted(mock_process_with_exception):
     # Hopefully this test will be unnecessary with um2nc standalone.
     # Test that the "Variable can not be processed" error arising from time
     # series inputs is excepted.
     allowed_error_message = esm1p5_convert.ALLOWED_UM2NC_EXCEPTION_MESSAGES[
         "TIMESERIES_ERROR"
     ]
     mock_process_with_exception(allowed_error_message)
-    with pytest.warns(Warning, match="Unable to convert"):
-        esm1p5_convert.convert_fields_file_list(["fake_file"], "fake_nc_write_dir")
+    fake_file_path = Path("fake_file")
+
+    _, failed = esm1p5_convert.convert_fields_file_list(
+        [fake_file_path], "fake_nc_write_dir")
+
+    assert failed[0][0] == fake_file_path
+
+    # TODO: Testing the exception part of the reported failures will be easier
+    # once um2nc specific exceptions are added.
 
 
-def test_convert_fields_file_list_generic_error(mock_process_with_exception):
+def test_convert_fields_file_list_fail_generic(mock_process_with_exception):
```

Comment:
Is this the test for when the conversion fails in a more critical fashion?

Reply:
Yeah! The main idea here is to have a list of exceptions which we want the driver to skip past when they occur without affecting the conversion of the rest of the files. E.g. ESM1.5 currently produces some timeseries output which will always cause the missing lon/lat coordinate error in `um2netcdf`, and we want the driver to just continue on past these files. For other exceptions though, we want the whole thing to crash which I'm testing here.

Reply:
Would it be worth renaming to something like `test_convert_fields_file_list_fail_critical` (and I guess adding a comment explaining the test...)